cmd/main.go

```diff
@@ -443,9 +442,14 @@ func main() {
 	hostStateMonitor.Start()
 	defer hostStateMonitor.Stop()
 
-	failOnError(addReleaseImagesToDBIfNeeded(db, releaseImagesArray, startupLeader, log), "error occured while adding configuration release images to the DB if needed")
-	openshiftReleaseSyncer := runOpenshiftReleaseSyncerIfNeeded(releaseSourcesArray, releaseImagesArray, db, log, releaseHandler, lead, startupLeader, failOnError)
-	defer stopOpenshiftReleaseSyncerIfNeeded(openshiftReleaseSyncer)
+	failOnError(
```

Comment:
in my mind this could be something like:
```
failOnError(lead.RunWithLeader(releaseSources.UpdateImagesIfEmpty(releaseSources, releaseImages)))
```
main difference is we leave lead stuff in main.

Reply:
we can have `RunWithLeader` in main, but then we are back in the problem of having the inside function in `main.go` (which doesn't make it shorter) or placing it in an external package (which doesn't really fit in `versions` or `releasesources`, so it means we beed to create a new package for it probably, which I don't know if it not overstepping the purpose of this PR. WDYT ?

Reply:
I would see those functions in another package, but even if we had those kind of function in main and all other logic specific for some package extracted, main would still be shorter than now